### PR TITLE
Bug - using a reference comparing operator for string comparison.

### PR DESCRIPTION
**Pull request by gitplaneta**
It's also worth considering using the Lombok project to generate boiler plate code like equals, hashcode and getters (https://projectlombok.org/features). Not sure though, whether it would interfere with the CodeGenProcessor class.

Signed-off-by: Radoslaw Busz <radoslaw.busz@gmail.com>

### CODE REVIEW

**Comment by vietj** [comment]
Hi, can you add a unit test for this please in HttpTest ?

**Comment by gitplaneta** [comment]
Hi, I've added the test for HttpServerOptions and it turned out that I need to add equality methods to additional classes to make it work. I've included additional tests for them as well.

**Comment by vietj** [comment]
thanks

File: src/main/java/io/vertx/core/net/JksOptions.java [hunk 1]
```diff
@@ -142,6 +142,30 @@ public JksOptions setValue(Buffer value) {
   }
 
   @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
```
Comment (vietj) [inline src/main/java/io/vertx/core/net/JksOptions.java]: please use braces

File: src/main/java/io/vertx/core/net/JksOptions.java [hunk 2]
```diff
@@ -142,6 +142,30 @@ public JksOptions setValue(Buffer value) {
...
+    if (!(o instanceof JksOptions)) return false;
```
Comment (vietj) [inline src/main/java/io/vertx/core/net/JksOptions.java]: please use braces

File: src/main/java/io/vertx/core/net/JksOptions.java [hunk 3]
```diff
@@ -142,6 +142,30 @@ public JksOptions setValue(Buffer value) {
...
+
+    JksOptions that = (JksOptions) o;
+
+    if (password != null ? !password.equals(that.password) : that.password != null) return false;
```
Comment (vietj) [inline src/main/java/io/vertx/core/net/JksOptions.java]: please use braces

File: src/main/java/io/vertx/core/net/JksOptions.java [hunk 4]
```diff
@@ -142,6 +142,30 @@ public JksOptions setValue(Buffer value) {
...
+    if (path != null ? !path.equals(that.path) : that.path != null) return false;
```
Comment (vietj) [inline src/main/java/io/vertx/core/net/JksOptions.java]: please use braces

File: src/main/java/io/vertx/core/net/JksOptions.java [hunk 5]
```diff
@@ -142,6 +142,30 @@ public JksOptions setValue(Buffer value) {
...
+    if (value != null ? !value.equals(that.value) : that.value != null) return false;
```
Comment (vietj) [inline src/main/java/io/vertx/core/net/JksOptions.java]: please use braces

**Comment by gitplaneta** [comment]
I've added braces into JksOptions, PemKeyCertOptions and PfxOptions.

**Comment by purplefox** [comment]
Hi, can you make sure you have signed the Eclipse CLA and the commits as described here:

https://github.com/vert-x3/wiki/wiki/How-to-contribute-to-Vert.x

Thanks

**Comment by gitplaneta** [comment]
I think I've done everything from the manual, the signed off commit:
`
commit 38982d45f1017ecfa346677c9f89de186b9f6408
Author: Radoslaw Busz <radoslaw.busz@gmail.com>
Date:   Tue Jun 2 15:59:36 2015 +0200

    Fixes HttpServerOptions.equals and adds required equals() methods into indirect dependecies
    
    Signed-off-by: Radoslaw Busz <radoslaw.busz@gmail.com>

`

the CLA witch matching email and my github account and my name
`The CLA we have on record for you will expire on 2018-05-12` 

Not sure what went wrong.

**Comment by purplefox** [comment]
Thanks.

**Pull request merged**